activerecord/test/cases/relation/where_test.rb

```diff
@@ -107,6 +107,10 @@ def test_where_with_tuple_syntax
       assert_equal [first_topic, third_topic].sort, Topic.where(key => conditions).sort
     end
 
+    def test_where_with_tuple_syntax_and_empty_array
```

Comment:
It's a little out of the scope of this PR but I think we will benefit from adding one more test and ensuring that 
`Topic.where([:id] => []).to_a` performs no queries similarly how `Topic.where(id: []).to_a` doesn't go to the db

Reply:
This one keeps me running in circles.

The whole thing hinges on a `WhereClause#contardiction?` returning true. And there are three ways to achieve that:

- Add an extra case recognising `"1=0"` to `#contardiction?`

    It's way too specific. What about "1=2"?

- Construct an arel node that satisfies existing case, translates to contradictory sql with `to_sql`, and use it in place of "1=0".

    `Arel::Node::In` would lie on its left side.
    `Arel::Node::Equality`, e.g. `Arel::Nodes::Equality.new(1, ActiveRecord::Relation::QueryAttribute.new("unboundable", -1, ActiveModel::Type::Integer.new(limit: 0)))` kinda works, but relies on something fairly unrelated.

- Implement a new arel node that's always contradictory, make `#contradiction?` aware of it, and use it as an initial value for reduction in predicate builder.

    `Arel::Nodes::Contradiction` that doesn't accepts any values during initialisation, gets discarded by visitors when it's a part of `Or` node, forces discard of the other side when it's a part of `And` node, turns into `IS TRUE` or `1=1` when negated, and so on.

    - As a shortcut, arel could provide an unboundable singleton for use in `Equality`, which would achieve all of the above, except for the ability to use it in `Or` and `And` nodes without polluting generated sql with unnecessary clauses.

I kinda lean towards the latter, as it implements an alluring identity...

Which one makes sense from outside of my head?

Reply:
I previously contemplated, but didn't merge, something maybe related: https://github.com/matthewd/rails/commit/792eb0442438698c49b98a9ab91e1d88217e321c

Reply:
That’s exactly it! And the names actually make sense, unlike my attempt.

Was there a blocker preventing merge?